app/src/main/java/com/example/memeitupapp/ui/listmemes/ListMemesFragment.kt

```diff
@@ -4,15 +4,63 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.Toast
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.LinearLayoutManager
 import com.example.memeitupapp.R
+import com.example.memeitupapp.data.entity.Meme
+import com.example.memeitupapp.ui.adapter.ListMemesAdapter
+import com.example.memeitupapp.util.Data
+import com.example.memeitupapp.util.Event
+import com.example.memeitupapp.util.Status
+import kotlinx.android.synthetic.main.fragment_list_memes.fragment_list_memes_progress_bar
+import kotlinx.android.synthetic.main.fragment_list_memes.fragment_list_memes_recycler_view
 
 class ListMemesFragment : Fragment() {
 
+    private lateinit var listMemesViewModel: ListMemesViewModel
+    private lateinit var listMemesAdapter: ListMemesAdapter
+
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? =
         inflater.inflate(R.layout.fragment_list_memes, container, false)
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        listMemesViewModel = ViewModelProvider(this).get(ListMemesViewModel::class.java)
+        listMemesViewModel.getMemesFromApi()
+        listMemesViewModel.mainState.observe(::getLifecycle, ::updateUI)
+    }
+
+    private fun showErrorMsg(errorMessage: String?) {
```

Comment:
why this can be null?

Reply:
because of the event class, that can return null